cmd/lakectl/cmd/doctor.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+var (
+	ErrCredential       = errors.New("credential error")
+	ErrWrongEndpointURI = errors.New("wrong endpoint_uri error")
+	ErrUnknownConfig    = errors.New("unknown configuratioin error")
+)
+
+// doctorCmd represents the doctor command
+var doctorCmd = &cobra.Command{
+	Use:   "doctor",
+	Short: "Run a basic diagnosis of the LakeFS configuration",
+	Run: func(cmd *cobra.Command, args []string) {
+		err := ListRepositoriesAndAnalyze(cmd.Context())
+		if err == nil {
+			Fmt("LakeFS doctor could not find any configuration issues\n")
+			return
+		}
+		configFileName := viper.GetViper().ConfigFileUsed()
+		Fmt("It looks like you have a problem with your `%v` file.\n", configFileName)
+		switch {
+		case errors.Is(err, ErrCredential):
+			Fmt("It is possible that the `access_key_id` or `secret_access_key` you supplied are wrong.\n")
```

Comment:
I don't understand.  Didn't we already report an error on l.78?

This is a bit odd.  I would just return an error from ListRepositoriesAndAnalyze, and format everything here.  E.g. if your error is a `struct` with fields `Message` and `Detail`, you can format them really nicely with the utils code here.  (It doesn't even need to be an error, I guess.)  But formatting part of the output here and part there makes it harder to read (and maybe also write) things like l.83, where we *don't* emit a message.

Reply:
Agree, do you think also the part of going over the file and trying to get what might be the problem should be in `ListRepositoriesAndAnalyze`?